pkg/nftlabs/nft.go

```diff
@@ -50,40 +52,58 @@ func (sdk *NftModule) MintBatch(meta []MintNftMetadata) ([]NftMetadata, error) {
 	if sdk.main.getSignerAddress() == common.HexToAddress("0") {
 		return nil, &NoSignerError{typeName: "nft"}
 	}
+	return sdk.MintBatchTo(sdk.main.getSignerAddress().String(), meta)
+}
 
-	var wg sync.WaitGroup
-	ch := make(chan string)
+func (sdk *NftModule) MintBatchTo(to string, meta []MintNftMetadata) ([]NftMetadata, error) {
+	if sdk.main.getSignerAddress() == common.HexToAddress("0") {
+		return nil, &NoSignerError{typeName: "nft"}
+	}
 
-	for _, asset := range meta {
-		wg.Add(1)
-		go func(meta interface{}) {
-			log.Printf("Uploading collection meta %v\n", meta)
-			defer wg.Done()
+	ch := make(chan [2]interface{})
 
-			uri, err := sdk.main.getGateway().Upload(meta, sdk.Address, sdk.main.getSignerAddress().String())
+	for index, asset := range meta {
```

Comment:
So this loop uploads the NFT metadata in batch. We actually have a method for this, but its not generalized yet. In the meantime, you can use it as a reference for how you can better design this loop:

https://github.com/nftlabs/nftlabs-sdk-go/blob/master/pkg/nftlabs/collection.go#L239

A few notes:
- If there is an issue uploading one of the NFTs metadata, we currently skip that asset. This is problematic, and we (the NFTLabs team) will make a decision soon on how we want this to behave for `v1.0.0`
- You're currently retrying the upload (which is great), but I'm not sure if we want this behavior just yet. We plan to generalize the `UploadBatch` function in the gateway which will have some sort of retry policy.
- Given that this isn't a stable version yet, I'm happy to merge your PR assuming you've tested the function. We may change it before releasing `v1.0.0`, but this is a great place to start.